src/users/repository/users.repository.ts

```diff
@@ -0,0 +1,174 @@
+            filterObject.userAttributes[key] = {
+              startsWith: value,
+              mode: 'insensitive',
+            };
+            return filterObject;
+          },
+          { userAttributes: {} },
+        )
+      : null;
+    const where = {
+      role: Role.USER,
+      ...(user.role !== Role.ADMIN && { active: true }),
+      ...(filter && { ...filter }),
+    };
+
+    return { take, skip, orderBy, where };
+  }
+
+  async find(query: FindUserDto, user: User): Promise<FindResponse<User>> {
+    const { take, skip, where, orderBy } = this.prepareFindQuery(query, user);
+
+    const [count, data] = await this.prismaService.$transaction([
+      this.prismaService.user.count({
+        where,
+      }),
+
+      this.prismaService.user.findMany({
+        skip,
+        take,
+        where,
+        include: { userAttributes: true },
+        ...(orderBy && { orderBy }),
+      }),
+    ]);
+
+    return { data, meta: { skip, take, count } };
+  }
+
+  async findOne(
+    id: number,
+  ): Promise<
+    Omit<
+      User,
+      | 'password'
+      | 'createdAt'
+      | 'updatedAt'
+      | 'userAttributesId'
+      | 'organisationAttributesId'
+    >
+  > {
+    const user = await this.prismaService.user.findUnique({
+      where: { id },
+      select: {
+        userAttributes: true,
+        organisationAttributes: true,
+        id: true,
+        role: true,
+        email: true,
+        active: true,
+        profilePhotoKey: true,
+      },
+    });
+
+    if (!user) {
+      throw new NotFoundException();
+    }
+
+    return user;
+  }
+
+  async update(id: number, updateUserDto: UpdateUserDto) {
+    const user = await this.prismaService.user.findUnique({
+      where: { id },
+      select: { role: true },
+    });
+
+    if (!user) {
+      throw new NotFoundException();
+    }
+
+    const userData = { active: updateUserDto.active };
+
+    const userAttributesData = {
+      firstname: updateUserDto.firstname,
+      lastname: updateUserDto.lastname,
+      city: updateUserDto.city,
+      phone: updateUserDto.phone,
+      type: updateUserDto.type,
+    };
+
+    const organisationAttributesData = {
+      name: updateUserDto.name,
+      street: updateUserDto.street,
+      oib: updateUserDto.oib,
+      city: updateUserDto.city,
+    };
+
+    return this.prismaService.user.update({
+      where: { id },
+      data: {
+        ...userData,
+        ...(user.role === Role.USER && {
+          userAttributes: { update: { ...userAttributesData } },
+        }),
+        ...(user.role === Role.ORGANISATION && {
+          organisationAttributes: { update: { ...organisationAttributesData } },
+        }),
+      },
+      select: {
+        userAttributes: true,
+        organisationAttributes: true,
+        id: true,
+        email: true,
+        active: true,
+      },
+    });
+  }
+
+  async remove(id: number): Promise<Pick<Certificate, 'key'>[] | undefined> {
```

Comment:
@aarbanas Usually when we have an repository which ties to an entity we should only return such entities, void, boolean etc.

A example that comes to my mind is that if you had to delete a user but want to send him an email that his account is deleted it's much easier to do such logic knowing that remove returns a User entity.